components/gfx/paint_context.rs

```diff
@@ -424,14 +471,24 @@ impl<'a> PaintContext<'a> {
                 if x0 > x1 {
                     mem::swap(&mut p0, &mut p1);
                 }
+                if rotated_axes {
+                    p0 = rotate_axes(p0, false);
+                    p1 = rotate_axes(p1, false);
+                }
                 (Some(p0), Some(p1))
             },
             (Some(x0), None) => {
```

Comment:
Is it just me or can this be written `(Some(x0), None) | (None, Some(x0)) => { ... }`?